Add IL2001-IL2060 trimmer warnings (#25842)
dotnet/docs: docs/core/deploying/trim-warnings/il2039.md

---
title: "IL2039: Invalid `action` value on resource element in a substitution file"
description: "Learn about trim warning IL2039: SubstitutionResourceAction"
ms.date: 08/25/2021
ms.topic: reference
author: mateoatr
ms.author: matorre
f1_keywords:
  - "IL2039"
---

> # IL2039: Invalid `action` value on resource element in a substitution file"

Reviewer: ```suggestion
# IL2039: Invalid `action` value on resource element in a substitution file
```

## Cause

> The action value passed to the `action` argument of a `resource` element in a substitution file does not have a valid value.

Reviewer 2: ```suggestion
The value passed to the `action` argument of a `resource` element in a substitution file is not valid.
```
